### ISSUE TITLE
How to ensure findOneById bypasses the cache and gets the data from the database? 

### ISSUE THREAD

**Issue by doronrk** [GraphQLGuide/apollo-datasource-mongodb#38]
There doesn't seem to be a way to ensure that `findOneById` gets its value directly from the database. If TTL is unset, then the read will not populate the cache after fetching the document, but it will _always_ try to get the key from the cache regardless of the value of TTL.

Is there a particular reason why this behavior is not supported? Sometimes when a use-case isn't supported, it makes me think I'm barking up the wrong tree. However, it seems reasonable to want to ensure that a given query gets the latest value from mongodb, even if a subsequent load was issued with a non-zero TTL.

Thoughts?

**Comment by doronrk** [comment]
Let's say you have two different gql resolvers. Resolver A comes under very heavy load and can return slightly stale values. Resolver B is hit rarely, but must return the latest value. 

Resolver B still wants to take advantage of the DataLoader semantics which guarantee that at most one db-fetch is issued for a particular ID, but this one db-fetch must come from mongodb, not Redis. 

Meanwhile, Resolver A is constantly causing Resolver B to get cache hits from Redis, because `findOneById` unconditionally checks redis first and returns early.

**Comment by lorensr** [comment]
Makes sense! Pro/cons of A vs adding an option like `findOneById(id, { bypassCache: true })`?

**Comment by doronrk** [comment]
Thanks for the fast response @lorensr ! 

I think the API you proposed is reasonable, but there is one drawback. Hypothetically, a subclass could want to do any combination of bypassCacheOnRead: yes/no, writeToCache: yes/no. 

I think the drawback is that some future person like me might open another issue asking for this. If you let the subclass access the dataloader, this provides the most flexibility. 

Thoughts?

**Comment by lorensr** [comment]
Sounds good, would appreciate PR for A, thanks ☺️

**Comment by doronrk** [comment]
Cool. I'm living the startup life at the moment, but will try and get to this.
